File: ios/PlankViewManager.m
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(PlankViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(onTime, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onText, RCTDirectEventBlock)
RCT_EXTERN_METHOD(updateFromManager:(nonnull NSNumber *)node)

@end

File: ios/PlankViewManager.swift
import Foundation

@objc(PlankViewManager)
class PlankViewManager: RCTViewManager {
  override func view() -> UIView! {
    return PlankView()
  }

  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc func updateFromManager(_ node: NSNumber) {
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view(forReactTag: node) as! SquatCounterView
      component.update()
    }
  }
}
